mage/src/mage/quick_test_runner.clj

```diff
@@ -167,20 +170,34 @@
 (defn go
   "Interactively select directories to run tests against."
   [{:keys [arguments options] :as _parsed}]
-  (let [tests (setup-test-files arguments options)]
-    (if (and (backend/nrepl-open?)
-             ;; No testing against the mage nrepl!
-             (not= :bb (backend/nrepl-type)))
-      (do
-        (println "Running tests via ⏩🏎️✨" (c/green (c/bold "THE REPL")) "✨🏎️⏪.")
-        (run-tests-over-nrepl tests options))
+  (let [tests (setup-test-files arguments options)
+        port (:port options)
+        nrepl-open? (backend/nrepl-open? port)
+        nrepl-type (backend/nrepl-type port)]
+    (u/debug (pr-str ["INFO" {:port port :nrepl-open? nrepl-open? :nrepl-type nrepl-type}]))
+    (cond
+      (not nrepl-open?)
       (do
+        (println (c/red "Unable to find a backend nrepl."))
         (println "Running via " (c/bold (c/magenta "the command line")) "."
-                 (c/red " This is " (c/bold "SLOW") " and " (c/bold "NOT RECCOMENDED!! "))
+                 (c/red " This is " (c/bold "SLOW") " and " (c/bold "NOT RECOMMENDED!! "))
                  "Please consider starting a backend \nFor quicker test runs, use: " (c/magenta "  clj -M:test:dev:ee:ee-dev:drivers:drivers-dev:dev-start"))
-        (println "\n" (banner
-                       {:font               bling.fonts.drippy/drippy
-                        :text               "Please open a REPL!"
-                        :gradient-direction :to-top
-                        :gradient-colors    [:magenta :red]}))
-        (run-tests-cli tests)))))
+        (println "\n" (banner {:text "Please open a REPL!"
+                               :font bling.fonts.drippy/drippy
+                               :gradient-direction :to-right
+                               :gradient-colors [:magenta :red]}))
+        (run-tests-cli tests))
+      (and nrepl-open? (= :clj nrepl-type))
+      (do
+        (println "Running tests via ⏩🏎️✨" (c/green (c/bold "THE REPL")) "✨🏎️⏪.")
+        (run-tests-over-nrepl tests options))
+      (and nrepl-open? (= :bb nrepl-type))
+      (println "You have a babashka repl open, and its port is in .nrepl-port.\n"
+               "Either use -p <backend port>, or put the clojure repl port into .nrepl-port to run tests via the backend repl.")
+
+      :else
+      (throw (ex-info "Unknown issue."
```

Comment:
this can happen if you have a cljs nrepl open at your port. I will update the message a bit